bluebottle/time_based/admin.py

```diff
@@ -404,10 +415,122 @@ class TimeContributionAdmin(ContributionChildAdmin):
     fields = ['contributor', 'slot_participant', 'created', 'start', 'end', 'value', 'status', 'states']
 
 
+class SlotWidget(TextInput):
+
+    template_name = 'admin/widgets/slot_widget.html'
+
+
+class ParticipantSlotForm(ModelForm):
+    checked = BooleanField(label=_('Participating'), initial=True, required=False)
+
+    def __init__(self, *args, **kwargs):
+        super(ParticipantSlotForm, self).__init__(*args, **kwargs)
+        initial = kwargs.get('initial', '')
+        slot = ''
+        if initial:
+            slot = initial['slot']
+        instance = kwargs.get('instance', '')
+        if instance:
+            slot = instance.slot
+        self.fields['slot'].label = _('Slot')
+        self.fields['slot'].widget = SlotWidget(attrs={'slot': slot})
+
+    class Meta:
+        model = SlotParticipant
+        fields = ['slot', 'checked']
+
+    def save(self, commit=True):
+        self.is_valid()
+        if not self.cleaned_data['checked']:
+            self.instance = None
+        else:
+            return super().save(commit)
+
+
+class ParticipantSlotFormSet(BaseInlineFormSet):
+
+    def __init__(self, *args, **kwargs):
+        if 'data' not in kwargs:
+            instance = kwargs['instance']
+            new = []
+            for slot in instance.activity.slots.exclude(slot_participants__participant=instance).all():
+                new.append({
+                    'slot': slot,
+                    'checked': False
+                })
+
+            kwargs.update({'initial': new})
+        super(ParticipantSlotFormSet, self).__init__(*args, **kwargs)
+
+    def save_new_objects(self, commit=True):
```

Comment:
Please specifiy what is overridden. 

Wasn't it the plan to transition unchecked objects instead of deleting them?